data/spells/scripts/support/levitate.lua

```diff
@@ -0,0 +1,36 @@
+function onCastSpell(creature, variant)
+	local currentPosition = creature:getPosition()
+	local destinationPosition = creature:getPosition()
+	destinationPosition:getNextPosition(creature:getDirection())
+
+	local returnValue = RETURNVALUE_NOTPOSSIBLE
+	local parameter = variant:getString():lower()
+	if parameter == "up" and currentPosition.z ~= 8 then
+		local tile = Tile(currentPosition.x, currentPosition.y, currentPosition.z - 1)
+		if not tile or not tile:getGround() and not tile:hasFlag(TILESTATE_IMMOVABLEBLOCKSOLID) then
+			tile = Tile(destinationPosition.x, destinationPosition.y, destinationPosition.z - 1)
+
+			if tile and tile:getGround() and not tile:hasFlag(bit.bor(TILESTATE_IMMOVABLEBLOCKSOLID, TILESTATE_FLOORCHANGE)) then
+				returnValue = Game.moveCreature(creature, tile, bit.bor(FLAG_IGNOREBLOCKITEM, FLAG_IGNOREBLOCKCREATURE))
+			end
+		end
+	elseif parameter == "down" and currentPosition.z ~= 7 then
+		local tile = Tile(destinationPosition)
+		if not tile or not tile:getGround() and not tile:hasFlag(TILESTATE_BLOCKSOLID) then
+			tile = Tile(destinationPosition.x, destinationPosition.y, destinationPosition.z + 1)
+
+			if tile and tile:getGround() and not tile:hasFlag(bit.bor(TILESTATE_IMMOVABLEBLOCKSOLID, TILESTATE_FLOORCHANGE)) then
```

Comment:
I'll and rebase and take a final look at this.
Last time I tried removing duplicated code I did something wrong, but I'll try again.